library/src/main/java/com/getstream/sdk/chat/view/MessageListView.java

```diff
@@ -147,13 +149,25 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
                     if (!hasScrolledUp) {
                         viewModel.setHasNewMessages(false);
                     }
-                    viewModel.setMessageListScrollUp(currentLastVisible + 1 < lVPosition);
+                    // delay of 100 milliseconds to prevent the effect when the keyboard is displayed.
+                    postDelayed(() -> {
+                        viewModel.setMessageListScrollUp(!lockScrollUp && currentLastVisible + 1 < lVPosition);
+                        lVPosition = currentLastVisible;
+                    }, 100);
                     fVPosition = currentFirstVisible;
-                    lVPosition = currentLastVisible;
                     viewModel.setThreadParentPosition(lVPosition);
                 }
             }
         });
+
+        // Lock for 500 milliseconds setMessageListScrollUp when the keyboard is displayed.
+        this.addOnLayoutChangeListener((View view, int left, int top, int right, int bottom,
```

Comment:
@adrian09h if this is really needed, please add more information here (sounds like an hack though to do this btw)